File: 프로그래머스/lv1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
def get_gcd(n1, n2):
    while True:
        if not (n1 % n2):
            return n2
        n1, n2 = n2, n1 % n2

def solution(n, m):
    answer = list()
    gcd = get_gcd(max(n, m), min(n, m))
    answer.append(gcd)
    answer.append(m * n // gcd)

    return answer

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.cpp
#include <vector>

using namespace std;

int get_gcd(int n, int m) {
    while (true) {
        if (n % m == 0) break;
        int temp = n % m;
        n = m;
        m = temp;
    }
    
    return m;
}

vector<int> solution(int n, int m) {
    vector<int> answer;
    int gcd = get_gcd(max(n, m), min(n, m));
    
    answer.emplace_back(gcd);
    answer.emplace_back(n * m / gcd);
    
    return answer;
}